charts/skypilot/templates/api-deployment.yaml

```diff
@@ -52,19 +58,20 @@ spec:
           # Any local changes to the config.yaml file will be overwritten by the contents of the configmap.
           cp /tmp/config.yaml /root/.sky/config.yaml
           {{- end }}
-          sky api start --deploy && tail -f /root/.sky/api_server/server.log
+          exec python -m sky.server.server --deploy --host 0.0.0.0 --port 46580
```

Comment:
I think we should try to use `sky api start` since the CLI may implement additional logic in the future and we want to use a consistent method to start the cluster. Redundant health checks should be ok (and even desirable, since this health check will fail sooner and can give more detailed error messages).

Reply:
Kind of disagree - the k8s pod should not try to do healthchecking or other logic that we may want in the convenient user-facing CLI.
IMO it definitely makes sense to have to entrypoints:
1. `sky api start` - start the server, if it's not running, background it, etc. I can run this on my laptop or on a remote server I'm manually ssh'd into.
2. `python -m sky.server.server` - Just shut up and run the server, I know what I'm doing. Run this from k8s, systemd service, etc.

It's a pretty clear delineation and it should be obvious what logic goes where.

Reply:
Hmm, I think we should have an argument for `sky api start` to run foreground, instead of calling the underlying interface directly. Mainly for maintainence: the arguments we passed into the `sky.server.server` module could subject to change, which could be easy to cause divergence if we have to maintain two separate parts. Also, sticking with a single set of API will make it cleaner. : )

See: https://docs.ray.io/en/latest/cluster/cli.html#cmdoption-ray-start-block

Reply:
`--foreground` sounds good to me. Curious though why doesn't `tail -f /root/.sky/api_server/server.log` work anymore?

Reply:
> Curious though why doesn't tail -f /root/.sky/api_server/server.log work anymore?

Did tail work fine previously? AFAIK, it should be flaky since tail only outputs the last 10 lines by default so we get complete log only when the server prints less than 10 lines before we start tailing.

Reply:
If `tail` is the only concern, we can also just set a `tail -n 10000 -f ...` to get all output?

Reply:
you can do `tail -n+0 -f ...` - this is always how i tail logs in ~/sky_logs

Reply:
I agree we should have a unified entry-point and `tail -n+0 -f ...` looks good to me for the tail problem. I drafted another PR to focus this single change: https://github.com/skypilot-org/skypilot/pull/4812

However, running API server in foreground is still relevant as it simplifies the deployment on k8s. Some points I can think:

- enables quick restart upon API server process exiting, instead of waiting for liveness probe;
- remove the additional log file and tail process in the container;

The priority might be low, will work later or until users explicitly ask for this.

Thanks! @cg505 @romilbhardwaj @Michaelvll